### PR TITLE
gperftools: drop an option that seems no longer supported

### LINKED ISSUE #109903: gperftools: can't be installed in mac OS (blocking mongodb)
**Describe the bug**
I can't install mongodb in mac OS. I get the following errors when using nix shell

```
checking how clang reports undeclared, standard C functions... configure: error: in `/private/tmp/nix-build-gperftools-2.8.drv-0/gperftools-2.8':
configure: error: cannot detect from compiler exit status or warnings
See `config.log' for more details
builder for '/nix/store/31v0wys0dz9clxyvjx3lc3jrnwnl3iqh-gperftools-2.8.drv' failed with exit code 1
cannot build derivation '/nix/store/zgxzf15h0dbbp9rr04kklgaflqk6pd20-mongodb-3.4.24.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/z25i9ykzh4s75l6q1dffqyvdl9nnyckk-lorri-keep-env-hack-nix-shell.drv': 1 dependencies couldn't be built
error: build of '/nix/store/z25i9ykzh4s75l6q1dffqyvdl9nnyckk-lorri-keep-env-hack-nix-shell.drv' failed
```

and the following if I do a vanilla install `nix-env -i mongodb`
```
checking how clang reports undeclared, standard C functions... configure: error: in `/private/tmp/nix-build-gperftools-2.8.drv-0/gperftools-2.8':
configure: error: cannot detect from compiler exit status or warnings
See `config.log' for more details
builder for '/nix/store/31v0wys0dz9clxyvjx3lc3jrnwnl3iqh-gperftools-2.8.drv' failed with exit code 1
cannot build derivation '/nix/store/zgxzf15h0dbbp9rr04kklgaflqk6pd20-mongodb-3.4.24.drv': 1 dependencies couldn't be built

```

**To Reproduce**
try installing mongodb by `nix-env -i mongodb` and observe the error.
**Expected behavior**
Should be able to install mongodb on mac os. 

**Screenshots**

**Additional context**

**Notify maintainers**
@kevincox @mohe2015 @timokau @dasJ @Mic92 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.2.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixpkgs-20.09pre236005.2b7c0dcdaab"`
 - channels(ashkanaleali): `"darwin, home-manager, nixpkgs-21.03pre264143.7d71001b796, nixpkgs-unstable-21.03pre264143.7d71001b796"`
 - nixpkgs: `/Users/ashkanaleali/.nix-defexpr/channels/nixpkgs`

### PR DESCRIPTION
**Pull request by vcunat**
> warning: unknown warning option '-Wno-aligned-allocation-unavailable';
>   did you mean '-Wno-debug-compression-unavailable'? [-Wunknown-warning-option]

- - -
Only darwin gets rebuilt, but it needs to be tested.

### CODE REVIEW

**Comment by vcunat** [comment]
It's de-facto revert of https://github.com/NixOS/nixpkgs/pull/36572/commits/6a560f499c5a1038a573ea961ef1d1166e1b3677

**Comment by vcunat** [comment]
We [were](https://nix-cache.s3.amazonaws.com/log/dghh3ygpkmd3v3hpyvypr0mrn82ghw4c-gperftools-2.8.drv) getting the warning even previously, but before it wasn't leading to this fatal problem.  Most likely that was triggered by the autoconf update &ndash; default, we still keep 269 and could switch to it selectively.

File: pkgs/development/libraries/gperftools/default.nix [hunk 1]
```diff
@@ -34,7 +34,7 @@ stdenv.mkDerivation rec {
   '';
 
   NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gperftools/default.nix:36]:
```suggestion
  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin
```

stdenv.lib is deprecated.

Comment (vcunat) [reply pkgs/development/libraries/gperftools/default.nix:36]: OK, I'd fix all the places in this file at once.  Can you link the deprecation?

Comment (SuperSandro2000) [reply pkgs/development/libraries/gperftools/default.nix:36]: https://github.com/NixOS/nixpkgs/commit/c06b2b3d671da4847f950e7e9041920b40eca0bf

You can search that yourself.

**Comment by prusnak** [comment]
I confirm that removing `-Wno-aligned-allocation-unavailable` from `NIX_CFLAGS_COMPILE` fixes the build of `gperftools`.

File: pkgs/development/libraries/gperftools/default.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/development/libraries/gperftools/default.nix:36]: Thanks :-)

**Pull request merged**